src/main/java/org/broadinstitute/hellbender/engine/FeatureWalker.java

```diff
@@ -79,7 +78,7 @@ private void initializeDrivingFeatures() {
      * Subclasses can override to provide their own behavior but default implementation should be suitable for most uses.
      */
     @Override
-    public void traverse() {
+    public final void traverse() {
```

Comment:
This `final` conflicts with the "Subclasses can override to provide their own behavior" comment above. We typically don't make `traverse()` final to allow for specializations of walker traversals (like `TwoPassVariantWalker` overriding `traverse()` from `VariantWalker`). I think the comment should be modified to read:

"You should only override traverse() if you are writing a new walker base class in the engine package that extends this class. It is not meant to be overridden by tools outside of the engine package."

While you're at it, can you make the same change to the comments for `traverse()` in the other walker base classes (and remove the finals).

Reply:
Fixed.